components/common/src/templating/package.rs

```diff
@@ -99,39 +101,43 @@ pub struct Pkg {
-                        svc_var_path: fs::svc_var_path(&package.ident.name),
-                        svc_pid_file: fs::svc_pid_file(&package.ident.name),
-                        svc_user,
-                        svc_group,
-                        env: Env::new(&package)?,
-                        deps: package.tdeps()?,
-                        exposes: package.exposes()?,
-                        exports: package.exports()?,
-                        path: package.installed_path.clone(),
-                        ident: package.ident.clone(),
-                        origin: package.ident.origin.clone(),
-                        name: package.ident.name.clone(),
-                        version: package.ident
-                                        .version
-                                        .clone()
-                                        .expect("No package version in PackageInstall"),
-                        release: package.ident
-                                        .release
-                                        .clone()
-                                        .expect("No package release in PackageInstall") };
+        let pkg =
+            Pkg { svc_path: fs::svc_path(&package.ident.name),
+                  svc_config_path: fs::svc_config_path(&package.ident.name),
+                  svc_config_install_path: fs::svc_config_install_path(&package.ident.name),
+                  svc_data_path: fs::svc_data_path(&package.ident.name),
+                  svc_files_path: fs::svc_files_path(&package.ident.name),
+                  svc_run: fs::svc_path(&package.ident.name).join("run"),
+                  svc_static_path: fs::svc_static_path(&package.ident.name),
+                  svc_var_path: fs::svc_var_path(&package.ident.name),
+                  svc_pid_file: fs::svc_pid_file(&package.ident.name),
+                  svc_user,
+                  svc_group,
+                  env: Env::new(&package)?,
+                  deps: package.tdeps()?,
+                  exposes: package.exposes()?,
+                  exports: package.exports()?,
+                  path: package.installed_path.clone(),
+                  ident: package.ident.clone(),
+                  origin: package.ident.origin.clone(),
+                  name: package.ident.name.clone(),
+                  version: package.ident
+                                  .version
+                                  .clone()
+                                  .expect("No package version in PackageInstall"),
+                  release: package.ident
+                                  .release
+                                  .clone()
+                                  .expect("No package release in PackageInstall"),
+                  shutdown_signal: package.shutdown_signal()?.unwrap_or_else(Default::default),
+                  shutdown_timeout: package.shutdown_timeout()?.unwrap_or_else(Default::default) };
```

Comment:
I completely agree about pushing error handling down into `PackageInstall`, but I think that is going to be a piece of work by itself, and may have far-reaching impact. I don't think that should be a part of this PR.

Reply:
Can we get an issue filed and do it as a direct follow up? If not, I think the likelihood of this getting fixed is low.

Reply:
@baumanj We can file an issue, but how direct a follow-up it would be will need some research. I started to look into this a while ago, but I think it had a much bigger footprint that you might otherwise think.

Reply:
Created issue #6572.